src/AccessDefinitions/Metadata/AccessDefinitionPropertyMetadata.php

```diff
@@ -14,48 +14,50 @@
 class AccessDefinitionPropertyMetadata extends PropertyMetadata {
 
   /**
-   * @var array|null
+   * @var AccessDefinitionMetadata|null
    */
-  public $readRoles;
+  public $canReadEveryone;
 
   /**
-   * @var array|null
+   * @var AccessDefinitionMetadata|null
    */
-  public $writeRoles;
+  public $canWriteEveryone;
 
   /**
-   * @var string|null
+   * @var AccessDefinitionMetadata|null
    */
-  public $readMethod;
+  public $canReadOwner;
 
   /**
-   * @var string|null
+   * @var AccessDefinitionMetadata|null
    */
-  public $writeMethod;
-
-  /**
-   * @var array|null
-   */
-  public $readScopes;
-
-  /**
-   * @var array|null
-   */
-  public $writeScopes;
+  public $canWriteOwner;
 
   /**
    * @param array $data
    *
    * @return $this
    */
   public function setPropertyMetadataFromConfigData(array $data): self {
```

Comment:
Can we make this method more generic? Seems like we're nearing a point where maintainability becomes an issue.

Reply:
Not sure what you mean by "more generic". But I do agree that the readability of this method is not great.